envoy/templates/ingress.yaml

```diff
@@ -0,0 +1,45 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "envoy.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```

Comment:
The beta1 of ingress is be removed in 1.22, and I don't think we need apiversion for `extensions,` as upstream does not support apiversion for `extenstions`.

https://kubernetes.io/blog/2021/08/04/kubernetes-1-22-release-announcement/